### PR TITLE
[9293] Log JAXRS exception handled by OT ExceptionMapper

### PR DESCRIPTION
**Pull request by andymc12**
This change resolves issue #9293.

The OT code currently handles exceptions from the JAX-RS application by mapping them to responses with a 500 Internal Server Error status code and a header with an exception key.  By doing this, the JAX-RS runtime does not log the exception - assuming it to be handled.

In order to log the exception so that the user is still notified of otherwise-unhandled exceptions, this change logs the exception as a warning.

### CODE REVIEW

File: dev/com.ibm.ws.opentracing.1.1/resources/com/ibm/ws/opentracing/resources/Opentracing.nlsprops [hunk 1]
```diff
@@ -74,6 +74,11 @@ OPENTRACING_NO_TRACERFACTORY=CWMOT0008E: OpenTracing cannot track JAX-RS request
 OPENTRACING_NO_TRACERFACTORY.explanation=OpenTracing will not function properly because an OpentracingTracerFactory class was not provided. JAX-RS requests will execute normally, but they will not be tracked.
 OPENTRACING_NO_TRACERFACTORY.useraction=For more information, see the documentation about how to enable OpenTracing distributed tracing.
 
+OPENTRACING_UNHANDLED_JAXRS_EXCEPTION=CWMOT0009W: The OpenTracing exception mapper has detected and is handling an unhandled exception from the JAX-RS application.
+OPENTRACING_UNHANDLED_JAXRS_EXCEPTION.explanation=The JAX-RS-based application has experienced and exception that is not handled within the application. The OpenTracing code will handle it by logging the exception stack trace and returning a 500 Internal Server Error response.
```
Comment (Charlotte-Holt) [inline dev/com.ibm.ws.opentracing.1.1/resources/com/ibm/ws/opentracing/resources/Opentracing.nlsprops]: Change to
"The JAX-RS-based application experienced an exception that is not handled within the application. The OpenTracing code handles it by logging the exception stack trace and returning a 500 Internal Server Error response."

File: dev/com.ibm.ws.opentracing.1.1/resources/com/ibm/ws/opentracing/resources/Opentracing.nlsprops [hunk 2]
```diff
@@ -74,6 +74,11 @@ OPENTRACING_NO_TRACERFACTORY=CWMOT0008E: OpenTracing cannot track JAX-RS request
...
+OPENTRACING_UNHANDLED_JAXRS_EXCEPTION.useraction=This may be acceptable behavior.  To handle the exception differently, the application developer can add an ExceptionMapper that handles the exception and maps it to a different response.
```
Comment (Charlotte-Holt) [inline dev/com.ibm.ws.opentracing.1.1/resources/com/ibm/ws/opentracing/resources/Opentracing.nlsprops]: Change to
"This behavior might be acceptable. To handle the exception differently, the application developer can add an ExceptionMapper interface that handles the exception and maps it to a different response."
* Note just one space between the sentences.

File: dev/com.ibm.ws.opentracing.1.1/resources/com/ibm/ws/opentracing/resources/Opentracing.nlsprops [hunk 3]
```diff
@@ -74,6 +74,11 @@ OPENTRACING_NO_TRACERFACTORY=CWMOT0008E: OpenTracing cannot track JAX-RS request
 OPENTRACING_NO_TRACERFACTORY.explanation=OpenTracing will not function properly because an OpentracingTracerFactory class was not provided. JAX-RS requests will execute normally, but they will not be tracked.
 OPENTRACING_NO_TRACERFACTORY.useraction=For more information, see the documentation about how to enable OpenTracing distributed tracing.
 
+OPENTRACING_UNHANDLED_JAXRS_EXCEPTION=CWMOT0009W: The OpenTracing exception mapper has detected and is handling an unhandled exception from the JAX-RS application.
```
[the remark this answers is not in the document]
Comment (andymc12) [reply dev/com.ibm.ws.opentracing.1.1/resources/com/ibm/ws/opentracing/resources/Opentracing.nlsprops]: Done - thanks for the review!

File: dev/com.ibm.ws.opentracing.1.1/resources/com/ibm/ws/opentracing/resources/Opentracing.nlsprops (hunk 1 above)
Comment (andymc12) [reply dev/com.ibm.ws.opentracing.1.1/resources/com/ibm/ws/opentracing/resources/Opentracing.nlsprops]: Done, thanks.

File: dev/com.ibm.ws.opentracing.1.1/resources/com/ibm/ws/opentracing/resources/Opentracing.nlsprops (hunk 2 above)
Comment (andymc12) [reply dev/com.ibm.ws.opentracing.1.1/resources/com/ibm/ws/opentracing/resources/Opentracing.nlsprops]: Done, thanks.

**Comment by andymc12** [comment]
Hi @fmhwong 

> Thanks for fixing this! Do you mind to add the change to the 1.0 project too?

Done.  Thanks for the review.

New #build with changes from code review comments.

File: dev/com.ibm.ws.opentracing.1.1/resources/com/ibm/ws/opentracing/resources/Opentracing.nlsprops (hunk 3 above)
[the remark this answers is not in the document]
Comment (Charlotte-Holt) [reply dev/com.ibm.ws.opentracing.1.1/resources/com/ibm/ws/opentracing/resources/Opentracing.nlsprops]: Thanks! ID review label added.

**Comment by andymc12** [comment]
Last build failed with an infrastructure issue... Previous build (before code review changes) had some failures because of the new warning appearing in the logs.  I've updated the FATs to so that they will now expect those warnings.  New #build

**Comment by andymc12** [comment]
#build

**Pull request merged**